---
title: 'Example data for TCGA Workflow: Analyze cancer genomics and epigenomics data using Bioconductor packages'
author: Tiago C. Silva, Antonio Colaprico, Catharina Olsen, Fulvio Dâ€™Angelo, Gianluca
  Bontempi Michele Ceccarelli , and Houtan Noushmehr
date: "`r Sys.Date()`"
output:  BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{'Example data for TCGA Workflow: Analyze cancer genomics and epigenomics data using Bioconductor packages'}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

# Data Introduction

This package provides a dataset for those wishing to try out the
[TCGA Workflow: Analyze cancer genomics and epigenomics data using Bioconductor packages](https://f1000research.com/articles/5-1542/v2) [@10.12688/f1000research.8923.2]. 
The data in this package are a subset of the TCGA data
for LGG (Lower grade glioma) and GBM (Glioblastoma multiforme) samples.

# Loading the data

```{r, fig.show='hold'}
library(TCGAWorkflowExampleData)
data("elmer.example")
data("GBMnocnvhg19")
data("GBMIllumina_HiSeq")
data("LGGIllumina_HiSeq")
data("met.20.samples.GBM.LGG.chr9")
```

# Data creation

The following commands were used to create the data included with this package.

## Copy number variations (CNVs) 

The following code will download 
segmented CNV from SNP array (Affymetrix Genome-Wide Human SNP Array
6.0) for 20 GBM samples.

```{r , eval=FALSE, include=TRUE, results='asis'}
library(TCGAbiolinks)
query.gbm.nocnv <- GDCquery(project = "TCGA-GBM",
                            data.category = "Copy number variation",
                            legacy = TRUE,
                            file.type = "nocnv_hg19.seg",
                            sample.type = c("Primary solid Tumor"))
# to reduce time we will select only 20 samples
query.gbm.nocnv$results[[1]] <- query.gbm.nocnv$results[[1]][1:20,]

GDCdownload(query.gbm.nocnv, chunks.per.download = 100)

gbm.nocnv <- GDCprepare(query.gbm.nocnv, save = TRUE, save.filename = "GBMnocnvhg19.rda")
```

## Gene expression data

The following code will download 20 LGG and 20 GBM samples that have both 
gene expression data. The Gene expression data is the raw expression signal for expression of a gene.

```{r , eval=FALSE, include=TRUE, results='asis'}
query <- GDCquery(project = "TCGA-GBM",
                  data.category = "Gene expression",
                  data.type = "Gene expression quantification",
                  platform = "Illumina HiSeq", 
                  file.type  = "results", 
                  sample.type = c("Primary solid Tumor"),
                  legacy = TRUE)
# We will use only 20 samples to make the example faster
query$results[[1]] <-  query$results[[1]][1:20,]                  
GDCdownload(query)
gbm.exp <- GDCprepare(query, save = TRUE, summarizedExperiment = TRUE, save.filename = "GBMIllumina_HiSeq.rda")

query <- GDCquery(project = "TCGA-LGG",
                  data.category = "Gene expression",
                  data.type = "Gene expression quantification",
                  platform = "Illumina HiSeq", 
                  file.type  = "results", 
                  sample.type = c("Primary solid Tumor"),
                  legacy = TRUE)
# We will use only 20 samples to make the example faster
query$results[[1]] <-  query$results[[1]][1:20,]
GDCdownload(query)
lgg.exp <- GDCprepare(query, save = TRUE, summarizedExperiment = TRUE, save.filename = "LGGIllumina_HiSeq.rda")
```


## DNA methylation and Gene expression data

The following code will select 10 LGG and 10 GBM samples that have both 
DNA methylation  and gene expression data. This objects will be then prepared 
to the format accept by the Biocondcutor package 
`r Biocpkg("ELMER")`([link])(http://bioconductor.org/packages/release/bioc/html/ELMER.html).
The DNA methylation will have only the chromossome 9 probes in order to make the example 
of the workflow faster. For a real analysis, all chromossomes should be used.
The Gene expression data is the normalized results for expression of a gene.


```{r , eval=FALSE, include=TRUE, results='asis'}
#----------- 8.3 Identification of Regulatory Enhancers   -------
library(TCGAbiolinks)
# Samples: primary solid tumor w/ DNA methylation and gene expression
matched_met_exp <- function(project, n = NULL){
    # get primary solid tumor samples: DNA methylation
    message("Download DNA methylation information")
    met450k <- GDCquery(project = project,
                        data.category = "DNA methylation",
                        platform = "Illumina Human Methylation 450",
                        legacy = TRUE, 
                        sample.type = c("Primary solid Tumor"))
    met450k.tp <-  met450k$results[[1]]$cases
    
    # get primary solid tumor samples: RNAseq
    message("Download gene expression information")
    exp <- GDCquery(project = project,
                    data.category = "Gene expression",
                    data.type = "Gene expression quantification",
                    platform = "Illumina HiSeq", 
                    file.type  = "normalized_results", 
                    sample.type = c("Primary solid Tumor"),
                    legacy = TRUE)
    exp.tp <- exp$results[[1]]$cases
    # Get patients with samples in both platforms
    patients <- unique(substr(exp.tp,1,15)[substr(exp.tp,1,12) %in% substr(met450k.tp,1,12)] )
    if(!is.null(n)) patients <- patients[1:n] # get only n samples
    return(patients)
}
lgg.samples <- matched_met_exp("TCGA-LGG", n = 10)
gbm.samples <- matched_met_exp("TCGA-GBM", n = 10)
samples <- c(lgg.samples,gbm.samples)

#-----------------------------------
# 1 - Methylation
# ----------------------------------
query.met <- GDCquery(project = c("TCGA-LGG","TCGA-GBM"),
                      data.category = "DNA methylation",
                      platform = "Illumina Human Methylation 450",
                      legacy = TRUE, 
                      barcode = samples)
GDCdownload(query.met)
met <- GDCprepare(query.met, save = FALSE)
met <- subset(met,subset = as.character(GenomicRanges::seqnames(met)) %in% c("chr9"))
save("met.20.samples.GBM.LGG.chr9.rda")

met.elmer <- TCGAprepare_elmer(met, platform = "HumanMethylation450")

#-----------------------------------
# 2 - Expression
# ----------------------------------
query.exp <- GDCquery(project = c("TCGA-LGG","TCGA-GBM"),
                     data.category = "Gene expression",
                     data.type = "Gene expression quantification",
                     platform = "Illumina HiSeq", 
                     file.type  = "normalized_results", 
                     legacy = TRUE, 
                     barcode =  samples)
GDCdownload(query.exp)
exp.lgg <- GDCprepare(query.exp, save = FALSE)
exp.elmer <- TCGAprepare_elmer(exp.lgg, platform = "IlluminaHiSeq_RNASeqV2")
save(exp.elmer, met.elmer, gbm.samples, lgg.samples, file = "elmer.example.rda", compress = "xz")
```



# Session info
```{r sessionInfo, results='asis', echo=FALSE}
pander::pander(sessionInfo(), compact = FALSE)
```
